src/main/java/io/stargate/sgv2/jsonapi/service/schema/tables/ApiRegularIndex.java

```diff
@@ -112,59 +83,183 @@ public ApiRegularIndex create(
-        // Only text and ascii fields can have the text analysis options specified
-        if (optionsDesc != null) {
-          var anyPresent =
-              Options.ASCII.isPresent(optionsDesc.ascii())
-                  || Options.CASE_SENSITIVE.isPresent(optionsDesc.caseSensitive())
-                  || Options.NORMALIZE.isPresent(optionsDesc.normalize());
-
-          if (anyPresent) {
-            throw SchemaException.Code.UNSUPPORTED_TEXT_ANALYSIS_FOR_DATA_TYPES.get(
-                errVars(
-                    tableSchemaObject,
-                    map -> {
-                      map.put(
-                          "allColumns",
-                          errFmtApiColumnDef(tableSchemaObject.apiTableDef().allColumns()));
-                      map.put("unsupportedColumns", errFmt(apiColumnDef));
-                    }));
-          }
-        }
-        // nothing to update in the cqlOptions for these indexes
-      } else {
-        // text and ascii fields can have the text analysis options specified
-        Options.ASCII.putOrDefaultStringable(
-            indexOptions, optionsDesc == null ? null : optionsDesc.ascii());
-        Options.CASE_SENSITIVE.putOrDefaultStringable(
-            indexOptions, optionsDesc == null ? null : optionsDesc.caseSensitive());
-        Options.NORMALIZE.putOrDefaultStringable(
-            indexOptions, optionsDesc == null ? null : optionsDesc.normalize());
+      // Rule 4
+      String indexFunction =
+          apiColumnDef.type().typeName() == ApiTypeName.MAP
+              ? ApiIndexFunction.ENTRIES.cqlFunction
+              : ApiIndexFunction.VALUES.cqlFunction;
+      indexOptions.put("indexFunction", indexFunction);
+
+      return new ApiRegularIndex(
+          indexIdentifier, ApiIndexType.COLLECTION, columnIdentifier, indexOptions);
+    }
+
+    /**
+     * Populate the indexOptions map by pulling values from optionsDesc. indexOptions will be used
+     * to append the options in createIndex cql statement.
+     */
+    private void populateIndexOptionsMap(
+        Map<String, String> indexOptions,
+        RegularIndexDefinitionDesc.RegularIndexDescOptions optionsDesc) {
+      Options.ASCII.putOrDefaultStringable(
+          indexOptions, optionsDesc == null ? null : optionsDesc.ascii());
+      Options.CASE_SENSITIVE.putOrDefaultStringable(
+          indexOptions, optionsDesc == null ? null : optionsDesc.caseSensitive());
+      Options.NORMALIZE.putOrDefaultStringable(
+          indexOptions, optionsDesc == null ? null : optionsDesc.normalize());
+    }
+
+    /**
+     * Only text and ascii datatypes can be analyzed. <br>
+     * Only text and ascii in the collection datatype can be analyzed. It works for values(list),
+     * values(set), keys(map), values(map). Note, not for entries(map). <br>
+     * The method will error out as UNSUPPORTED_TEXT_ANALYSIS_FOR_DATA_TYPES if rules are violated.
+     */
+    private void validateAnalyzableDatatypes(
```

Comment:
nit: none of these indexes are analyzed indexes, that term is only used the analyzed text indexes that specify the index_analyzer property